### PR TITLE
[DocumentIntelligence] Updated CHANGELOG and added migration guide

### PR DESCRIPTION
**Pull request by kinelski**
Changes to the README and to samples will be included in the next PR.

### CODE REVIEW

File: sdk/documentintelligence/Azure.AI.DocumentIntelligence/CHANGELOG.md [hunk 1]
```diff
@@ -14,3 +14,146 @@
 
 ### Features Added
 - This is the first preview of the `Azure.AI.DocumentIntelligence` package, targeting version 2023-10-31-preview of the Document Intelligence service.
+
+**Note: Document Intelligence is a rebranding of Form Recognizer. While this package and `Azure.AI.FormRecognizer` target the same Azure service, `Azure.AI.DocumentIntelligence` is not backward compatible and should be seen as a new library. See the [Migration Guide][migration_guide] for detailed instructions on how to update application code from `Azure.AI.FormRecognizer` version `4.X.X`. The following list of changes uses version 4.1.0 for comparison to assist during migration.**
```
Comment (pallavit) [inline sdk/documentintelligence/Azure.AI.DocumentIntelligence/CHANGELOG.md:18]: Is this the 2nd preview of the SDK?  I do not see the version update?

When I looked at the below list added property, removed field etc. I assume this is between 2 versions. Let me know if I am misinterpreting the changes.

File: sdk/documentintelligence/Azure.AI.DocumentIntelligence/MigrationGuide.md [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+
+- In `Azure.AI.DocumentIntelligence`, we have `DocumentIntelligenceClient` and `DocumentIntelligenceAdministrationClient` which support API version `2023-10-31-preview` and higher.
+- Some client methods have been renamed. See the [CHANGELOG][changelog] for an exhaustive list of changes.
+
+Creating new clients in `Azure.AI.FormRecognizer`:
+```C#
+string endpoint = "<endpoint>";
+string apiKey = "<apiKey>";
+var credential = new AzureKeyCredential(apiKey);
+
```
Comment (pallavit) [inline sdk/documentintelligence/Azure.AI.DocumentIntelligence/MigrationGuide.md:69]: Should this not be using older FormRecognizerClient?

**Review by pallavit — APPROVED** [review]

File: sdk/documentintelligence/Azure.AI.DocumentIntelligence/CHANGELOG.md (hunk 1 above)
Comment (pallavit) [reply sdk/documentintelligence/Azure.AI.DocumentIntelligence/CHANGELOG.md:18]: Discussed offline with @kinelski, this is intentional

File: sdk/documentintelligence/Azure.AI.DocumentIntelligence/MigrationGuide.md (hunk 2 above)
Comment (kinelski) [reply sdk/documentintelligence/Azure.AI.DocumentIntelligence/MigrationGuide.md:69]: We're comparing `DocumentIntelligenceClient` (from the new `Azure.AI.DocumentIntelligence` library) with `DocumentAnalysisClient` (from the old `Azure.AI.FormRecognizer` library).

**Pull request merged**